server/handler/fetcher.go

```diff
@@ -37,26 +39,50 @@ type ConfigFetcher struct {
 }
 
 func (cf *ConfigFetcher) ConfigForRepositoryBranch(ctx context.Context, client *github.Client, owner, repository, branch string) FetchedConfig {
+	retries := 0
+	delay := 1 * time.Second
+	ticker := time.NewTicker(delay)
+	defer ticker.Stop()
+	for {
+		c, err := cf.Loader.LoadConfig(ctx, client, owner, repository, branch)
+		fc := FetchedConfig{
+			Source: c.Source,
+			Path:   c.Path,
+		}
 
-	c, err := cf.Loader.LoadConfig(ctx, client, owner, repository, branch)
-	fc := FetchedConfig{
-		Source: c.Source,
-		Path:   c.Path,
-	}
+		if err != nil {
+			if !os.IsTimeout(err) {
```

Comment:
I think it would be worth re-trying other errors too, for instance if GitHub returns a 500 error or something.

Reply:
For this one, I guess the change needs to be done in [palantir/go-githubapp](https://github.com/palantir/go-githubapp) since [getFileContents](https://github.com/palantir/go-githubapp/blob/f40ba9b0944c4b7252c5a64fdea7f6565b9f3704/appconfig/appconfig.go#L270C17-L270C18) receives a response and only propagates the error. In order to check for 500, there must either be an error type to check for on the caller side, or some other way to propagate 500 error somehow.

Reply:
The `go-github` library returns an error for any non-successful status code, so we have two options here without changing the upstream:

* Retry on any error from loading the content and accept that maybe that will sometimes retry things that will still fail
* Use `errors.As` to test for [github.ErrorResponse](https://pkg.go.dev/github.com/google/go-github/v67/github#ErrorResponse) and then check the status code in the response